// @shopify/cli/dist/chunk-DM56XRBC.js
import {
  runThemeDoctor
} from "./chunk-E7IWFBAC.js";
import {
  globalFlags
} from "./chunk-AVEWTIFC.js";
import {
  base_command_default
} from "./chunk-KQU2IAXJ.js";
import {
  canRunDoctorRelease
} from "./chunk-6O2ILC2V.js";
import {
  require_lib
} from "./chunk-EZZRDRCA.js";
import {
  cwd,
  resolvePath
} from "./chunk-H4D3T5OG.js";
import {
  __toESM,
  init_cjs_shims
} from "./chunk-PKR7KJ6P.js";

// src/cli/commands/doctor-release/theme/index.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var DoctorReleaseTheme = class _DoctorReleaseTheme extends base_command_default {
  static {
    this.description = "Run all theme command doctor-release tests";
  }
  static {
    this.hidden = !0;
  }
  static {
    this.flags = {
      ...globalFlags,
      path: import_core.Flags.string({
        char: "p",
        description: "The path to run tests in. Defaults to current directory.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      }),
      environment: import_core.Flags.string({
        char: "e",
        description: "The environment to use from shopify.theme.toml (required for store-connected tests).",
        env: "SHOPIFY_FLAG_ENVIRONMENT",
        required: !0
      }),
      store: import_core.Flags.string({
        char: "s",
        description: "Store URL (overrides environment).",
        env: "SHOPIFY_FLAG_STORE"
      }),
      password: import_core.Flags.string({
        description: "Password from Theme Access app (overrides environment).",
        env: "SHOPIFY_FLAG_PASSWORD"
      })
    };
  }
  async run() {
    if (!canRunDoctorRelease())
      return;
    let { flags } = await this.parse(_DoctorReleaseTheme);
    (await runThemeDoctor({
      path: flags.path,
      environment: flags.environment,
      store: flags.store,
      password: flags.password
    })).some((result) => result.status === "failed") && (process.exitCode = 1);
  }
};

export {
  DoctorReleaseTheme
};
