// @shopify/cli/dist/local-SJCC5BYA.js
import {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  canRunDoctorRelease,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  getThemeKitAccessDomain,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken
} from "./chunk-6O2ILC2V.js";
import "./chunk-6NVYATES.js";
import "./chunk-CTFDRWUN.js";
import "./chunk-7IK72W75.js";
import "./chunk-EZZRDRCA.js";
import "./chunk-KUJQ4AB6.js";
import "./chunk-UMUTXITN.js";
import "./chunk-UV5N2VL7.js";
import "./chunk-XE5EOEBL.js";
import "./chunk-Q2H6D2MH.js";
import "./chunk-G2ZZKGSV.js";
import "./chunk-H4D3T5OG.js";
import "./chunk-PKR7KJ6P.js";
export {
  alwaysLogAnalytics,
  alwaysLogMetrics,
  analyticsDisabled,
  canRunDoctorRelease,
  ciPlatform,
  cloudEnvironment,
  codespacePortForwardingDomain,
  codespaceURL,
  firstPartyDev,
  getThemeKitAccessDomain,
  gitpodURL,
  hasGit,
  homeDirectory,
  isCloudEnvironment,
  isDevelopment,
  isShopify,
  isTerminalInteractive,
  isUnitTest,
  isVerbose,
  macAddress,
  opentelemetryDomain,
  themeToken
};
